src/engine/SpatialJoin.cpp

```diff
@@ -418,13 +425,13 @@ Result SpatialJoin::baselineAlgorithm() {
                             rightJoinCol);
 
       // Ensure `maxDist_` constraint
-      if (dist.getDatatype() != Datatype::Int || maxDist_ < 0 ||
-          dist.getInt() > maxDist_) {
+      if (dist.getDatatype() != Datatype::Int || !maxDist.has_value() ||
```

Comment:
Doesn;t this lead to "always return 0 results if we have no maxDist" ? This looks wrong to me, isn't it
` || (maxDist.has_value() && maxDist.value() < dist)`